public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -111,11 +115,54 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return defaultValue;
   };
 
+  requestViaWorker(url: string, hasObjectResponse?: false): Promise<string[]>;
+  requestViaWorker(url: string, hasObjectResponse?: true): Promise<PromMetricsMetadata>;
+  requestViaWorker(url: string, hasObjectResponse?: boolean): Promise<string[] | PromMetricsMetadata>;
```

Comment:
Seems like this override is not needed if we have false/true already.

Reply:
Unfortunately, it is necessary for proper autocomplete support. If this line is removed, you will only see 2 out of the 3 possible signatures, as TS doesn't add the implementation signature automatically.

Reply:
I should maybe elaborate a bit more on this - the crucial thing is that it would work as it is used right now but if `requestViaWorker()` was to be called with a boolean variable as the second argument, you'd get:
`Argument of type 'boolean' is not assignable to parameter of type 'true'.ts(2769)`